lib/pure/asyncfutures.nim

```diff
@@ -196,6 +196,15 @@ proc add(callbacks: var CallbackList, function: CallbackFunc) =
         last = last.next
       last.next = newCallback
 
+proc complete*(future: FutureBase) =
+  ## Completes a base ``future``.
```

Comment:
Emphasise in the docs that this likely shouldn't be used unless you really know what you're doing :)